sprint-3/step-1: добавил роутинг и авторизацию

srsad opened the pull request.

Commit:
sprint-3/step-1: добавил роутинг и авторизацию

andryxxa93 commented on src/app/index.jsx:
```diff
@@ -0,0 +1,112 @@
+    }
+  }, [dispatch, hasAuth])
+
+  useEffect(() => {
+    setAppLoaded(true)
+    delete state?.backgroundLocation
+  }, [])
```
**Можно лучше**
Без этого кода тоже работает. Насколько он действительно нужен и насколько нужно appLoaded состояние?

srsad replied:
У меня локально (хром 118.0.5993.70) не работает без этого

andryxxa93 commented on src/pages/forgot-password/index.jsx:
```diff
@@ -0,0 +1,59 @@
+    email: '',
+  })
+
+  const handleChange = (e) =>{
```
**Можно лучше**

Можно сделать универсальный **кастомный** хук для контроля любого количества инпутов в любых формах:
```js
 export function useForm(inputValues) {
   const [values, setValues] = useState(inputValues);
 
   const handleChange = (event) => {
     const {value, name} = event.target;
     setValues({...values, [name]: value});
   };
   return {values, handleChange, setValues};
 }
```
    
Этот код помещают в отдельный файл `useForm.js` в папке `hooks` и импортируют функцию туда, где нужно контролировать инпуты

 И Вам не нужно будет теперь вручную создавать функции обработки инпутов и т д. Все будет в одной строчке кода:
    
 ```js
   const {values, handleChange, setValues} = useForm({});
 ```

andryxxa93 reviewed (changes requested):
Здравствуйте! Проделана большая работа 👍 

Сильные стороны работы:
- Реализована защита роутов
- Роуты помещены в константы
- Есть механизм обновления токена
- Хорошая работа с куками

Необходимо исправить:
- При логауте срабатывает сообщение об ошибке.

<img width="1439" alt="image" src="https://github.com/srsad/react-burger/assets/60173362/ba27d094-78a1-4e1e-9ca0-367314d25f42">

- Не правильно работает восстановление пароля. Эндпоинт на который нужно отсылать код с новым паролем неверный. Пожалуйста обратите ещё раз внимание на тех-задание.
<img width="556" alt="image" src="https://github.com/srsad/react-burger/assets/60173362/d93a817b-88ae-484e-9687-9e32c2ffc6cb">

**Можно лучше**
- Когда выбрана "История заказов" то подсвечивается ссылка "Профиль" хотя не должна.
<img width="925" alt="image" src="https://github.com/srsad/react-burger/assets/60173362/7aaf0a2c-862f-4ae4-9062-78e2fedb8fa7">


В pull request вы так же найдете комментарии с рекомендациями как **можно улучшить** код. Их делать не обязательно, однако в будущем они могут помочь избежать ошибок и сделать код лучше.

**Напомню, чтобы работа была принята нужно исправить все замечания помеченные как **необходимо исправить**.

Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка F12) перед отправкой на ревью. 

Удачных правок!

Commit:
sprint-3/step-1: исправил ошибку при выходе
поправил эндпоинт для смены пароля
добвил универсальный хук для работы с формами
избавился от лишнего кода

andryxxa93 reviewed (approved):
Супер, вы отлично поработали!
Поздравляю! Работа принята.
Желаю удачного продолжения обучения!
